.ci/lib.sh

```diff
@@ -314,7 +314,7 @@ gen_clean_arch() {
 	info "Remove installed kubernetes packages and configuration"
 	if [ "$ID" == ubuntu ]; then
 		sudo rm -rf /etc/systemd/system/kubelet.service.d
-		sudo apt-get remove kubeadm kubelet kubectl docker.io docker-ce containerd.io docker-ce-cli -y
+		sudo apt-get remove kubeadm kubelet kubectl docker.io docker-ce containerd docker-ce-cli -y
```

Comment:
Couple of queries:

- Should we try to remove both `containerd` and `continerd.io`?
- If `containerd*` isn't installed, this command will fail. And since this script might be sourced with `set -e`, we probably need to add `|| true` at the end. Alternatively, the script could query the name of the containerd package before trying to remove it (which sounds safer to me).

/cc @GabyCT, @chavafg.

Reply:
> * Should we try to remove both `containerd` and `continerd.io`?

@jodh-intel This is run for Ubuntu. It seems to me that `containerd.io` wasn't packaged on Ubuntu, ever. `containerd.io` was the package in RHEL/CentOS 7.
> * If `containerd*` isn't installed, this command will fail.

I just found out `apt-get remove` does not fail when a package is in index, but not installed. It does fail when it isn't in index. This will also fail on ppc64le & s390x since `docker-ce-cli` isn't referenced anywhere (`docker-ce` is referenced, but not installable, must use `docker.io`).
> And since this script might be sourced with `set -e`, we probably need to add `|| true` at the end. Alternatively, the script could query the name of the containerd package before trying to remove it (which sounds safer to me).

`"... docker* containerd*"` too broad?

Reply:
Hi @Jakob-Naucke.

> containerd.io wasn't packaged on Ubuntu, ever.

That implies this code never ran then, of if it did, it failed the job ;)

> `"... docker* containerd*"` too broad?

It might be. An `egrep` pattern of `^(containerd|containerd.io|docker|docker.io|...)$` might be safer if that looks reasonable (not sure about the `docker-ce` package names)?

/cc @GabyCT, @chavafg.

Reply:
> That implies this code never ran then, of if it did, it failed the job ;)

This is only run on "bare metal" CIs, which isn't a lot of them. Also, _as is_, that code is not `set -e`'d, but your concern that it might be is right.
> > `"... docker* containerd*"` too broad?
> 
> It might be. An `egrep` pattern of `^(containerd|containerd.io|docker|docker.io|...)$` might be safer if that looks reasonable (not sure about the `docker-ce` package names)?

Sure. Let me get back to you.

Reply:
@jodh-intel `download.docker.com` (which doesn't package for ppc64le/s390x) calls it `containerd.io`, that's where it's from. Excuse my s390x ignorance :)

Reply:
Yes, although this script isn't `set -e`, I think all users of this script are and `source` this file.